lld/Common/ErrorHandler.cpp

```diff
@@ -219,6 +219,7 @@ void ErrorHandler::reportDiagnostic(StringRef location, Colors c,
   }
   os << msg << '\n';
   errs() << buf;
+  sep = getSeparator(msg);
```

Comment:
Perhaps add a comment like

if msg contains a newline, ensure that the next diagnostic is preceded by a blank line separator.